graphql/src/query/ast.rs

```diff
@@ -64,11 +65,17 @@ pub fn get_argument_value<'a>(arguments: &'a [(Name, Value)], name: &Name) -> Op
 }
 
 /// Returns true if a selection should be skipped (as per the `@skip` directive).
-pub fn skip_selection(selection: &Selection) -> bool {
+pub fn skip_selection(selection: &Selection, variables: &HashMap<Name, Value>) -> bool {
     match get_directive(selection, "skip".to_string()) {
         Some(directive) => match get_argument_value(&directive.arguments, &"if".to_string()) {
             Some(val) => match val {
+                // Skip if @skip(if: true)
                 Value::Boolean(skip_if) => *skip_if,
+                // Also skip if @skip(if: $variable) where $variable is true
+                Value::Variable(name) => variables.get(name).map_or(false, |var| match var {
+                    Value::Boolean(v) => v.to_owned(),
+                    _ => false,
```

Comment:
Good catch about `@include`. I've implemented it correctly now and added variables to it. Tested in GraphQL Playground as well.

On `@skip` and `@include` showing up with errors in GraphiQL: That could be a bug on their end. GraphQL Playground happily accepts them.

On validation: Yep, before executing anything, the entire query, all its variables, arguments, types etc. should be validated. #325 tracks that. Shouldn't be too hard to do, so this may be something for early January.

Reply:
Everything settled then! Since it seems straightforward to test query resolution, could you also add tests for include and skip with variables?

Reply:
@leodasvacas Done, tests added!

Reply:
Great!